### PR TITLE
Suppress ExecutionContext flowing to timer callbacks where available.

### LINKED ISSUE #52: Suggestion: Suppress ExecutionContext flow on Timer created by PortableTimer
An [OpenTelemetry](https://github.com/open-telemetry/opentelemetry-dotnet) user brought this up via Gitter:

> A less obvious example of this occurring can happen with the Serilog logging library. If you use a sink that relies on periodic batching, the first attempt to log something will start a timer, I believe, to periodically send data. If that first call to log something happened when there's an available Activity.Current it will indefinitely hold onto that activity.

OpenTelemetry is using [System.Diagnostics.DiagnosticSource.Activity.Current](https://github.com/dotnet/runtime/blob/7162efa9505fbb7b497a4d0257f78c07b8a22124/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Current.net46.cs#L10), which is backed by an AsyncLocal.

What's going on is this code in `PortableTimer`:
https://github.com/serilog/serilog-sinks-periodicbatching/blob/313598fd6c0747489c0fdf3949238dd75e60dae0/src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/PortableTimer.cs#L43

Captures whatever Activity.Current happens to be set when the ctor is invoked because the [.NET Framework](https://referencesource.microsoft.com/#mscorlib/system/threading/timer.cs,545) & [.NET Core](https://github.com/dotnet/coreclr/blob/a9f3fc16483eecfc47fb79c362811d870be02249/src/System.Private.CoreLib/shared/System/Threading/Timer.cs#L438) Timers capture execution context by default.

Thought I would suggest updating the code to wrap the ctor in an `ExecutionContext.SuppressFlow()` because I don't think the capture is intentional and it might prevent other mysterious things from happening in the future to serilog users.

/cc @nrcventura

### PR DESCRIPTION
**Pull request by CodeBlanch**
Resolves #52 

/cc @nblumhardt @nrcventura

### CODE REVIEW

File: src/Serilog.Sinks.PeriodicBatching/Serilog.Sinks.PeriodicBatching.csproj [hunk 1]
```diff
@@ -20,7 +20,7 @@
     <RepositoryType>git</RepositoryType>
     <GenerateAssemblyVersionAttribute>false</GenerateAssemblyVersionAttribute>
     <!-- Don't reference the full NETStandard.Library -->
-    <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>    
+    <DisableImplicitFrameworkReferences Condition=" '$(TargetFramework)' == 'net45' ">true</DisableImplicitFrameworkReferences>
```
Comment (nblumhardt) [inline src/Serilog.Sinks.PeriodicBatching/Serilog.Sinks.PeriodicBatching.csproj]: I'm not sure this does anything at all on .NET Framework; can probably drop it completely, now.

Comment (CodeBlanch) [reply src/Serilog.Sinks.PeriodicBatching/Serilog.Sinks.PeriodicBatching.csproj]: @nblumhardt It definitely does _something_:

image

Whether or not that _something_ is important, not sure 😄 Happy to remove if you want, just please confirm for me.

Comment (nblumhardt) [reply src/Serilog.Sinks.PeriodicBatching/Serilog.Sinks.PeriodicBatching.csproj]: :-) on .NET Framework, these assemblies will be in the GAC, so there's no packaging overhead like there would have been for the earlier netstandard targets - should be "nothing important" 👍

Comment (CodeBlanch) [reply src/Serilog.Sinks.PeriodicBatching/Serilog.Sinks.PeriodicBatching.csproj]: Updated

**Review by nblumhardt — APPROVED** [review]

**Comment by nblumhardt** [comment]
Thanks! LGTM 👍 

I have a couple of things to check around target framework support, I'll push the button shortly :-)

**Comment by nblumhardt** [comment]
Thanks again; sorry about the delay 😅

**Pull request merged**